### PR TITLE
LPS-79950

### PR DESCRIPTION
**Pull request by jonmak08**
Hey @jbalsas,

Attached is an update for http://issues.liferay.com/browse/LPS-79950.

I have included the commits from Rob for the ```Liferay.lazy``` method to lazy load the AMD modules. It seems to work well, but in looking at the pull request, I saw there was still some discussion between @brunobasto, @yuchi, and @izaera about this.  I also saw that a ticket was not created for this, so I wanted to send it through to you to see what else needs to be done before we push everything in.

Please advise and let me know if there is anything else you need from me for this.  I have included the changes for the task portlet in here, but the task portlet throw's a ```NullPointer``` and has other styling/formatting issues, which I will create a ticket for.

Please let me know if you have any questions. Thanks!

### CODE REVIEW

**Comment by jonmak08** [comment]
Styling issues filed at https://issues.liferay.com/browse/LPS-80286

**Comment by jonmak08** [comment]
NullPointerException issue filed at https://issues.liferay.com/browse/LPS-80287

File: modules/apps/tasks/tasks-portlet/docroot/upcoming_tasks/view.jsp [hunk 1]
```diff
@@ -94,10 +94,10 @@
 	</div>
 </div>
 
-<aui:script>
+<aui:script use="liferay-plugin-mail">
```
Comment (jbalsas) [inline modules/apps/tasks/tasks-portlet/docroot/upcoming_tasks/view.jsp]: Why are we using `liferay-plugin-mail` here? Shouldn't this be `liferay-tasks`?

Comment (jonmak08) [reply modules/apps/tasks/tasks-portlet/docroot/upcoming_tasks/view.jsp]: Whoops probably a copy and paste typo

Comment (jonmak08) [reply modules/apps/tasks/tasks-portlet/docroot/upcoming_tasks/view.jsp]: Fixed with https://github.com/jbalsas/liferay-portal/pull/1190/commits/db1759c2f389bd9dc20a5031e1e3b1b584334da8

**Comment by jbalsas** [comment]
Hey @jonmak08! 

I've been looking at this, and it looks good to me. There's no LPS for the initial `lazy` implementation. Could you create one for it and push this forward?

Regarding the name, I honestly can't come up with anything else... maybe we could make it more obvious by calling it something like: 
- `Liferay.wrap`,
- `Liferay.createAsync`
- `Liferay.createDeferred`
- `Liferay.createLazy`
- `Liferay.defineLazy`

I don't think any of those are any better, what do you think?

Unless you have a better alternative, could you create an LPS for the first commit and send to Brian? If we come up with something better down the line, we can always change and keep `lazy` as an alias.

Thanks!

**Comment by jonmak08** [comment]
```Liferay.lazyLoad``` API sent to @brianchandotcom at https://github.com/brianchandotcom/liferay-portal/pull/58154.

**Comment by jonmak08** [comment]
New Pull Request: https://github.com/brianchandotcom/liferay-portal/pull/58208

**Pull request closed without merging**